packages/web3-utils/src/validation.ts

```diff
@@ -109,46 +110,36 @@ export const isTopicInBloom = isTopicInBloomValidator;
  */
 export const compareBlockNumbers = (blockA: Numbers, blockB: Numbers) => {
 	// string validation
-	if (
-		typeof blockA === 'string' &&
-		!(
-			blockA === 'genesis' ||
-			blockA === 'earliest' ||
-			blockA === 'pending' ||
-			blockA === 'latest'
-		)
-	)
-		throw new InvalidBlockError(blockA);
-	if (
-		typeof blockB === 'string' &&
-		!(
-			blockB === 'genesis' ||
-			blockB === 'earliest' ||
-			blockB === 'pending' ||
-			blockB === 'latest'
-		)
-	)
-		throw new InvalidBlockError(blockB);
+	if (blockA === 'genesis' || blockB === 'genesis')
+		throw new InvalidBlockError('Genesis tag not supported'); // for more specific error message
+	if (typeof blockA === 'string' && !isBlockTag(blockA)) throw new InvalidBlockError(blockA);
+	if (typeof blockB === 'string' && !isBlockTag(blockB)) throw new InvalidBlockError(blockB);
+
+	// Increasing order:  earliest, safe, (finalized ~ latest), pending
+	// safe vs block-num cant be compared as block number provided can be on left or right side of safe tag, until safe tag block number is extracted and compared
+
 	if (
 		blockA === blockB ||
-		((blockA === 'genesis' || blockA === 'earliest' || blockA === 0) &&
-			(blockB === 'genesis' || blockB === 'earliest' || blockB === 0))
-	)
+		((blockA === 'earliest' || blockA === 0) && (blockB === 'earliest' || blockB === 0))
+	) {
 		return 0;
-
-	// b !== a, thus a < b
-	if (blockA === 'genesis' || blockA === 'earliest') return -1;
-
-	// b !== a, thus a > b
-	if (blockB === 'genesis' || blockB === 'earliest') return 1;
-
-	if (blockA === 'latest') {
+	}
+	if (blockA === 'earliest') {
+		// b !== a, thus a < b
+		return -1;
+	}
+	if (blockB === 'earliest') {
+		// b !== a, thus a > b
+		return 1;
+	}
```

Comment:
```suggestion
	if (blockA === 'earliest' || blockA === 0) {
		// b !== a, thus a < b
		return -1;
	}
	if (blockB === 'earliest' || blockB === 0) {
		// b !== a, thus a > b
		return 1;
	}
```

Reply:
this was already catered in cases below, in case A or B is 0 and other was valid block tag,. but good to check earlier in logic.